src/common/config_parser.cpp

```diff
@@ -408,31 +419,34 @@ void ConfigParser::addOptionsTraining(cli::CLIWrapper& cli) {
   addSuboptionsTSV(cli);
 
   // data management options
-  cli.add<std::string>("--shuffle",
-      "How to shuffle input data (data: shuffles data and sorted batches; batches: "
-      "data is read in order into batches, but batches are shuffled; none: no shuffling). "
-      "Use with '--maxi-batch-sort none' in order to achieve exact reading order", "data");
-  cli.add<bool>("--no-shuffle",
-      "Shortcut for backwards compatiblity, equivalent to --shuffle none (deprecated)");
-  cli.add<bool>("--no-restore-corpus",
-      "Skip restoring corpus state after training is restarted");
-  cli.add<std::string>("--tempdir,-T",
-      "Directory for temporary (shuffled) files and database",
-      "/tmp");
-  cli.add<std::string>("--sqlite",
-      "Use disk-based sqlite3 database for training corpus storage, default"
-      " is temporary with path creates persistent storage")
-    ->implicit_val("temporary");
-  cli.add<bool>("--sqlite-drop",
-      "Drop existing tables in sqlite3 database");
+  if (mode_ != cli::mode::selfadaptive) {
```

Comment:
Can you add a short comment why these options are excluded in the marian-adaptive? I think `--no-restore-corpus`, etc. don't make much sense, and `--sqlite` is not supported for simplicity, right? What about `--shuffle`?

Reply:
Yeah, `--no-restore-corpus` doesn't make sense. I think `--sqlite` also doesn't make sense because the context sentences in the "training" files are only read once, used for a few iterations and then discarded. I don't see when in the training  process would be appropriate to create an sqlite database with these sentences other than to maybe create the database beforehand and use it in place of the file inputs? That sounds complicated to support, though.

`--shuffle`, however, would make sense to support, i guess. I think i left it out because your code originally left it out and i didn't want to investigate. Now that i look at it, though, i see that the `TextInput` class is used for train set reading and it apparently doesn't support shuffling:
```c++
void shuffle() override {}
```
I guess we could change `TextInput::shuffle`, bet i don't know what other code might rely on the current behavior.

Reply:
I'm inclined not to implement shuffle at this point. Unless you @snukky disagree.

Reply:
That's fine, thanks. Please left a comment in the code saying which options are not supported in marian-adaptive by design, and that shuffle might be supported in future.